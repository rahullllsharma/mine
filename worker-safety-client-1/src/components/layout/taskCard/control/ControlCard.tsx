import type { PropsWithClassName } from "@/types/Generic";
import cx from "classnames";

export type ControlProps = PropsWithClassName<{ label: string }>;

export default function Control({
  label,
  children,
  className = "",
}: ControlProps): JSX.Element {
  return (
    <div
      className={cx(
        "flex text-base text-neutral-shade-100 p-3 border-dashed border border-brand-gray-30 rounded",
        className
      )}
      data-testid="controlCard"
    >
      <span className="flex-1">{label}</span>
      {children}
    </div>
  );
}
