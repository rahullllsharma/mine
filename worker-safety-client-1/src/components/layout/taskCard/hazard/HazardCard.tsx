import type { PropsWithChildren, ReactNode } from "react";

type HazardCardContentProp = PropsWithChildren<{
  header?: ReactNode;
}>;

export default function HazardCard({
  header,
  children,
}: HazardCardContentProp): JSX.Element {
  return (
    <div
      className="text-base text-neutral-shade-100 mb-2"
      data-testid="hazardCard"
    >
      <header className="flex flex-1 text-brand-gray-80  py-2 pr-2 mr-1 font-semibold">
        {header}
      </header>
      {children}
    </div>
  );
}
