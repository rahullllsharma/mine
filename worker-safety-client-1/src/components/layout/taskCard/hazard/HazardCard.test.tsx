import { render, screen } from "@testing-library/react";
import ControlCard from "../control/ControlCard";
import HazardCard from "./HazardCard";

describe("HazardCard", () => {
  it("should render a label", () => {
    render(<HazardCard header="Lorem ipsum" />);
    const labelElement = screen.getByText("Lorem ipsum");
    expect(labelElement).toBeInTheDocument();
  });

  it("should render a custom header with a button", () => {
    const header = (
      <div>
        <button>Lorem ipsum</button>
      </div>
    );
    render(<HazardCard header={header} />);
    const labelElement = screen.getByRole("button");
    expect(labelElement).toBeInTheDocument();
  });

  describe("when we have a set of controls", () => {
    it("should render 3 control cards", () => {
      render(
        <HazardCard header="Lorem ipsum">
          <ControlCard label="Situational Jobsite Awareness" />
          <ControlCard label="Trained and Qualified" />
          <ControlCard label="Erection of Proper Barricades and Warning Signs" />
        </HazardCard>
      );
      const controlCardElements = screen.getAllByTestId("controlCard");
      expect(controlCardElements).toHaveLength(3);
    });

    it("should render a list of controls", () => {
      const { asFragment } = render(
        <HazardCard header="Lorem ipsum">
          <ControlCard label="Situational Jobsite Awareness" />
          <ControlCard label="Trained and Qualified" />
          <ControlCard label="Erection of Proper Barricades and Warning Signs" />
        </HazardCard>
      );
      expect(asFragment()).toMatchSnapshot();
    });
  });
});
