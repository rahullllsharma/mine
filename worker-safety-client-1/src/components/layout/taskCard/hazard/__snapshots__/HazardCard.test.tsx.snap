// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HazardCard when we have a set of controls should render a list of controls 1`] = `
<DocumentFragment>
  <div
    class="text-base text-neutral-shade-100 mb-2"
    data-testid="hazardCard"
  >
    <header
      class="flex flex-1 text-brand-gray-80 py-2 pr-2 mr-1 font-semibold"
    >
      Lorem ipsum
    </header>
    <div
      class="flex text-base text-neutral-shade-100 p-3 border-dashed border border-brand-gray-30 rounded"
      data-testid="controlCard"
    >
      <span
        class="flex-1"
      >
        Situational Jobsite Awareness
      </span>
    </div>
    <div
      class="flex text-base text-neutral-shade-100 p-3 border-dashed border border-brand-gray-30 rounded"
      data-testid="controlCard"
    >
      <span
        class="flex-1"
      >
        Trained and Qualified
      </span>
    </div>
    <div
      class="flex text-base text-neutral-shade-100 p-3 border-dashed border border-brand-gray-30 rounded"
      data-testid="controlCard"
    >
      <span
        class="flex-1"
      >
        Erection of Proper Barricades and Warning Signs
      </span>
    </div>
  </div>
</DocumentFragment>
`;
